proposed/longhorn/latest/README.md

```diff
@@ -9,22 +9,34 @@ Longhorn is 100% open source software. Project source code is spread across a nu
 ## Prerequisites
 
 1. Docker v1.13+
-2. Kubernetes v1.8+
+2. Kubernetes v1.8+ cluster with 1 or more nodes and Mount Propagation feature enabled. If your Kubernetes cluster was provisioned by Rancher v2.0.7+ or later, MountPropagation feature is enabled by default. [Check your Kubernetes environment now](https://github.com/rancher/longhorn#environment-check-script). If MountPropagation is disabled, legacy K8s FlexVolume driver will be deployed instead of the newer K8s CSI driver. Volumes backed by a Base Image will also be disabled.
 3. Make sure `curl`, `findmnt`, `grep`, `awk` and `blkid` has been installed in all nodes of the Kubernetes cluster.
 4. Make sure `open-iscsi` has been installed in all nodes of the Kubernetes cluster. For GKE, recommended Ubuntu as guest OS image since it contains `open-iscsi` already.
 
 ## Uninstall Longhorn
 
-In order to uninstall Longhorn, user need to remove all the volumes first:
-```
-kubectl -n longhorn-system delete lhv --all
-```
+**Do not attempt to immediately delete Longhorn App from Rancher UI. Longhorn propagates device mounts which can damage the host if not unmounted properly.** Follow this procedure instead.
 
-After confirming all the volumes are removed, then Longhorn can be easily uninstalled using:
+1. To prevent damage to the Kubernetes cluster, we recommend deleting all Kubernetes workloads using Longhorn volumes (Deployments, StatefulSets, DaemonSets, etc).
+
+2. Run the following commands.
 ```
-kubectl delete -f https://raw.githubusercontent.com/rancher/longhorn/master/deploy/longhorn.yaml
+NAMESPACE=longhorn-system
+kubectl -n ${NAMESPACE} delete volumes.longhorn.rancher.io --all
+kubectl -n ${NAMESPACE} delete engineimages.longhorn.rancher.io --all
+kubectl -n ${NAMESPACE} delete deployment.apps/longhorn-driver-deployer
+kubectl -n ${NAMESPACE} delete daemonset.apps/longhorn-csi-plugin
+kubectl -n ${NAMESPACE} delete statefulset.apps/csi-provisioner
+kubectl -n ${NAMESPACE} delete service/csi-provisioner
+kubectl -n ${NAMESPACE} delete daemonset.apps/longhorn-flexvolume-driver
+kubectl -n ${NAMESPACE} delete daemonset.apps/longhorn-manager
+kubectl -n ${NAMESPACE} get nodes.longhorn.rancher.io -o yaml | sed "s/\- longhorn.rancher.io//g" | kubectl apply -f -
+kubectl -n ${NAMESPACE} delete nodes.longhorn.rancher.io --all
+kubectl -n ${NAMESPACE} delete settings.longhorn.rancher.io --all
 ```
 
+3. Delete Longhorn App from Rancher UI. This deletes the remaining chart components.
+
```

Comment:
Add one section in the troubleshooting, tell the user how to clean up after hit the uninstall in the UI without properly cleaned up. There will always be users doing that, and they will ask every time. It's better for us to point them to some documents. They may need to manually remove finalizer etc.